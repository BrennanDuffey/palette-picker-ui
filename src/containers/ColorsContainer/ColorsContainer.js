import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setColors, toggleLocked } from '../../actions';
import Color from '../../components/Color/Color.js';
import { Link } from 'react-router-dom';

class ColorsContainer extends Component {
  componentDidMount() {
    if(!this.props.colors[0].hexCode) {
      this.generateColors();
    };
  };

  randomColorGenerator = () => {
    let randomColor = "#000000".replace(/0/g, function() {
      return (Math.floor(Math.random() * 16)).toString(16);
    });
    return randomColor;
  }; 

  generateColors = () => {
    const colors = this.props.colors.map((color, index) => {
      if (color.isLocked) {
        return color;
      } else {
        return { isLocked: false, hexCode: this.randomColorGenerator(), id: index };
      };
    });
    this.props.setColors(colors);
  };

  render() {
    const colors = this.props.colors.map(color => (
      <Color {...color} toggleLocked={this.props.toggleLocked} />
    ));

    return (
      <div>
        <div className="colors">
          {colors}
        </div>
        <div className="button-container">
          <button onClick={this.generateColors}>GENERATE</button>
          {this.props.user.username && <Link to='/home/save'> <button>SAVE</button> </Link>}
        </div>
      </div>
    );
  };
};

export const mapStateToProps = (state) => ({
  colors: state.colors,
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  toggleLocked: (id) => dispatch(toggleLocked(id)),
  setColors: (colors) => dispatch(setColors(colors))
});

export default connect(mapStateToProps, mapDispatchToProps)(ColorsContainer);