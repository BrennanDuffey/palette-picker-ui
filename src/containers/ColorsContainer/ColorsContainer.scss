.ColorsContainer {
  .button-container {
    align-items: center;
    display: flex;
    flex-direction: column;
    margin: 15px 0 0 0;
  }

  button {
    background: $gray;
    border: none;
    color: white;
    font-size: 1.5em;
    font-weight: 800;
    margin: 15px 0 0 0;
    padding: 15px;
    width: 500px;

    &:hover {
      background: $lightGray;
    }
  }
}

.button {
  font-size: 1em;
}

.colors {
  display: flex;
  width: 100%;  
};
