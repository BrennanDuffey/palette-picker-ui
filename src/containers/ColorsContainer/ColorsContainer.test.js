import React from 'react';
import { shallow } from 'enzyme';
import { ColorsContainer } from './ColorsContainer';

describe('ColorsContainer', () => {
  let wrapper;
  let mockColors;
  let mockUser;
  let mockSetColors;

  beforeEach(() => {
    mockSetColors = jest.fn();
    mockColors = [
      {
        isLocked: false,
        hexCode: '#cc36de',
        id: 0
      },
      {
        isLocked: false,
        hexCode: '#e3caa7',
        id: 1
      },
      {
        isLocked: false,
        hexCode: '#8da892',
        id: 2
      },
      {
        isLocked: false,
        hexCode: '#3d9168',
        id: 3
      },
      {
        isLocked: false,
        hexCode: '#57798e',
        id: 4
      }
    ];
    mockUser = {}
    wrapper = shallow(
      <ColorsContainer setColors={mockSetColors} colors={mockColors} user={mockUser} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot on conditional rendering', () => {
    mockUser = {
      id: 2,
      username: 'something'
    };
    wrapper = shallow(
      <ColorsContainer colors={mockColors} user={mockUser} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call randomColorGenerator on generateColors', () => {
    wrapper.instance().randomColorGenerator = jest.fn()
    wrapper.instance().generateColors();
    expect(wrapper.instance().randomColorGenerator).toHaveBeenCalled();
  });

  it('should return new hexcodes on generate colors', () => {
    wrapper.instance().randomColorGenerator = jest.fn()
    wrapper.instance().generateColors();
    expect(mockSetColors).toHaveBeenCalled();
  });
});