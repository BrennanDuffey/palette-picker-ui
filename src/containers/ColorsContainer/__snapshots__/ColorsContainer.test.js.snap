// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorsContainer should match snapshot 1`] = `
<div>
  <div
    className="colors"
  >
    <Color
      hexCode="#cc36de"
      id={0}
      isLocked={false}
      key="#cc36de"
    />
    <Color
      hexCode="#e3caa7"
      id={1}
      isLocked={false}
      key="#e3caa7"
    />
    <Color
      hexCode="#8da892"
      id={2}
      isLocked={false}
      key="#8da892"
    />
    <Color
      hexCode="#3d9168"
      id={3}
      isLocked={false}
      key="#3d9168"
    />
    <Color
      hexCode="#57798e"
      id={4}
      isLocked={false}
      key="#57798e"
    />
  </div>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      GENERATE
    </button>
  </div>
</div>
`;

exports[`ColorsContainer should match snapshot on conditional rendering 1`] = `
<div>
  <div
    className="colors"
  >
    <Color
      hexCode="#cc36de"
      id={0}
      isLocked={false}
      key="#cc36de"
    />
    <Color
      hexCode="#e3caa7"
      id={1}
      isLocked={false}
      key="#e3caa7"
    />
    <Color
      hexCode="#8da892"
      id={2}
      isLocked={false}
      key="#8da892"
    />
    <Color
      hexCode="#3d9168"
      id={3}
      isLocked={false}
      key="#3d9168"
    />
    <Color
      hexCode="#57798e"
      id={4}
      isLocked={false}
      key="#57798e"
    />
  </div>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      GENERATE
    </button>
    <Link
      to="/home/save"
    >
       
      <button>
        SAVE
      </button>
       
    </Link>
  </div>
</div>
`;
